lib/oli_web/pow/author_context.ex

```diff
@@ -27,6 +28,43 @@ defmodule OliWeb.Pow.AuthorContext do
     |> maybe_delete_author_cached()
   end
 
+  @doc """
+  Overrides the default Pow.Ecto.Context `create`.
+  """
+  @impl true
+  def create(params) do
+    case Accounts.get_author_by_email(params["email"]) do
+      %Author{email: email} = author ->
```

Comment:
WDYT about separate this logic in small functions, something like that:  
def create(params) do
    case Accounts.get_author_by_email(params["email"]) do
      %Author{email: email} = author ->
        if author.email_confirmed_at do
          email_already_registered(email)
        else
          {:error, %{email: "has already been taken"}}
        end
  
      _nil ->
        create_new_author(params)
    end
  end
  
  
  
  defp email_already_registered(email) do
    Oli.Email.create_email(
      email,
      "Account already registered",
      "email_already_registered.html",
      %{
        url: Routes.authoring_pow_session_path(OliWeb.Endpoint, :new),
        forgot_password:
          Routes.authoring_pow_reset_password_reset_password_path(OliWeb.Endpoint, :new)
      }
    )
    |> Oli.Mailer.deliver_now()
    {:error, %{email: "has already been taken"}}
  end
  
  defp create_new_author(params) do
    %Author{}
    |> Author.changeset(params)
    |> Repo.insert()
    |> case do
      {:ok, author} ->
        {:ok, author}
  
      {:error, error} ->
        {:error, error}
    end
  end